src/main/scala/mesosphere/marathon/core/election/CuratorElectionStream.scala

```diff
@@ -242,15 +241,11 @@ object CuratorElectionStream extends StrictLogging {
       if (selfParticipantCount > 1) {
         throw new IllegalStateException(s"Multiple election participants have the same id: ${hostPort}. This is not allowed.")
       } else {
-        participants.find(_.isLeader).map(_.getId) match {
-          case Some(leader) if leader == hostPort =>
-            sq.offer(LeadershipState.ElectedAsLeader)
-          case None if participants.nonEmpty && retries < 3 =>
-            logger.info(s"Leader record was removed while querying participants; retrying after ${retries} retries")
-            emitLeader(loopId, retries + 1)
-          case otherwise =>
-            sq.offer(LeadershipState.Standby(otherwise))
+        val element = participants.find(_.isLeader).map(_.getId) match {
```

Comment:
can we find less generic name as element? What do you think about `newLeadershipState`

Reply:
so basically you removed the workaround you inserted before to make the tests more stable, right? I like that! :) It seemed a bit 'hacky' anyway...

Reply:
Yeah, in hindsight it was a mistake to try and make LeaderLatch completely deterministic. It backfired. The code was more complex and there are multiple ways that LeaderLatch can return no current leader. Better to just make the test more flexible.